### PR TITLE
Producer for genparticle quantities

### PR DESCRIPTION
**Pull request by harrypuuter**
This producer adds genparticle information for the two particles of the reco DiTauPair

- pt
- eta
- phi
- mass
- PDGID
- m_vis

### CODE REVIEW

File: config/config.py [hunk 1]
```diff
@@ -66,6 +67,8 @@ def build_config():
             BasicJetQuantities,
             BJetCollection,
             BasicBJetQuantities,
+            MTGenPair,
```
Comment (swozniewski) [inline config/config.py:70]: Maybe make MTGenPair part of GenDiTauPairQuantities?

File: pairselection.hxx [hunk 2]
```diff
@@ -16,7 +16,38 @@
...
+                  const std::string &genindex_particle1,
+                  const std::string &genindex_particle2,
+                  const std::string &genpair) {
+    auto getGenPair = [](const ROOT::RVec<int> recopair,
+                         const ROOT::RVec<int> genindex_particle1,
+                         const ROOT::RVec<int> genindex_particle2) {
+        ROOT::RVec<int> genpair = {-1, -1};
+        Logger::get("buildgenpair")->debug("existing DiTauPair: {}", recopair);
+        genpair[0] = genindex_particle1[recopair.at(0)];
+        genpair[1] = genindex_particle1[recopair.at(0)];
```
Comment (swozniewski) [inline pairselection.hxx:43]: fix indices

File: quantities.hxx [hunk 3]
```diff
@@ -186,6 +186,27 @@ auto isolation(auto df, const std::string &outputname, const int position,
...
+///
+/// \returns a dataframe with the new column
+
+auto pdgid(auto df, const std::string &outputname, const int position,
+           const std::string &pairname, const std::string &pdgidcolumn) {
+    return df.Define(
+        outputname,
+        [position](const ROOT::RVec<int> &pair, const ROOT::RVec<int> &pdgid) {
+            const int index = pair.at(position);
+            return pdgid[index];
```
Comment (swozniewski) [inline quantities.hxx:206]: my personal preference would be to do this in one go. I assume that the compiler optimizes it anyway, so this is not a strong request.

**Review by swozniewski — CHANGES_REQUESTED** [review]

File: quantities.hxx (hunk 3 above)
Comment (harrypuuter) [reply quantities.hxx:206]: I left it consistent to the other functions

**Review by swozniewski — APPROVED** [review]

**Pull request merged**